src/module_cache/WasmModuleCache.cpp

```diff
@@ -70,33 +60,39 @@ wasm::WAVMWasmModule& WasmModuleCache::getCachedModule(
         }
     }
 
-    // Stop now if we're just looking for the base cached module
-    if (specialKey == baseKey) {
+    // See if we're looking for a special snapshot
+    const std::string snapshotKey = msg.snapshotkey();
+    if (snapshotKey.empty()) {
         return cachedModuleMap[baseKey];
     }
 
-    // See if we already have the special cached module
-    if (getCachedModuleCount(specialKey) == 0) {
-        faabric::util::FullLock lock(mx);
-        if (cachedModuleMap.count(specialKey) == 0) {
-            // Get the base module and the special module
-            logger->debug("Creating new special zygote: {}", specialKey);
-            wasm::WAVMWasmModule& baseModule = cachedModuleMap[baseKey];
-            wasm::WAVMWasmModule& specialModule = cachedModuleMap[specialKey];
-
-            // Clone the special module from the base one
-            specialModule = baseModule;
-
-            // Restore the special module
-            specialModule.restoreFromState(specialKey, msg.snapshotsize());
-
-            // Write memory to fd
-            int fd = memfd_create(specialKey.c_str(), 0);
-            specialModule.writeMemoryToFd(fd);
+    if (!snapshotKey.empty()) {
+        // See if we already have the snapshot
+        if (getCachedModuleCount(snapshotKey) == 0) {
+            faabric::util::FullLock lock(mx);
+
+            if (cachedModuleMap.count(snapshotKey) == 0) {
+                // Get the base module and the special module
+                logger->debug("Creating new zygote from snapshot: {}",
+                              snapshotKey);
+                wasm::WAVMWasmModule& baseModule = cachedModuleMap[baseKey];
+                wasm::WAVMWasmModule& specialModule =
+                  cachedModuleMap[snapshotKey];
+
+                // Clone the special module from the base one
+                specialModule = baseModule;
+
+                // Restore the special module
+                specialModule.restoreFromState(snapshotKey, msg.snapshotsize());
+
+                // Write memory to fd to support copy-on-write cloning
+                int fd = memfd_create(snapshotKey.c_str(), 0);
+                specialModule.writeMemoryToFd(fd);
+            }
```

Comment:
This is basically just a rearrange to make it more readable and easier to step through